File: app/src/test/java/com/tjb/dwf/PinchGestureReceiverTest.java
package com.tjb.dwf;

import com.tjb.dwf.main.MainActivity;
import com.tjb.dwf.main.PinchGestureReceiver;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.mockito.runners.MockitoJUnitRunner;

import static org.mockito.Mockito.verify;

@RunWith(MockitoJUnitRunner.class)
public class PinchGestureReceiverTest {
    @Mock
    private MainActivity mainActivity;

    private PinchGestureReceiver pinchGestureReceiver;

    @Before
    public void setup() {
        pinchGestureReceiver = new PinchGestureReceiver();
        pinchGestureReceiver.installMainActivity(mainActivity);
    }

    @Test
    public void showOptions_then_showOptions() {
        // when
        pinchGestureReceiver.showOptions();

        // then
        // ...

        // verify
        verify(mainActivity).showOptions();
    }

    @Test
    public void showPicture_then_showPicture() {
        // when
        pinchGestureReceiver.showPicture();

        // then
        // ...

        // verify
        verify(mainActivity).showPicture();
    }
}

File: app/src/test/java/com/tjb/dwf/PinchGestureReceiverTest.kt
package com.tjb.dwf

import com.tjb.dwf.main.MainActivity
import com.tjb.dwf.main.PinchGestureReceiver
import io.mockk.mockk
import io.mockk.spyk
import io.mockk.verify
import org.apache.tools.ant.Main
import org.junit.Before
import org.junit.Test

class PinchGestureReceiverTest {
    private val pinchGestureReceiver = PinchGestureReceiver()

    private val mainActivity = mockk<MainActivity>(relaxed = true)

    @Before
    fun setup() {
        pinchGestureReceiver.installMainActivity(mainActivity);
    }

    @Test
    fun showOptions_then_showOptions() {
        // when
        pinchGestureReceiver.showOptions();

        // then
        // ...

        // verify
        verify {
            mainActivity.showOptions()
        }
    }

    @Test
    fun showPicture_then_showPicture() {
        // when
        pinchGestureReceiver.showPicture()

        // then
        // ...

        // verify
        verify {
            mainActivity.showPicture()
        }
    }
}
